spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestMetadataTableReadableMetrics.java

```diff
@@ -190,6 +191,7 @@ private GenericRecord createNestedRecord(Long longCol, Double doubleCol) {
   }
 
   @Test
+  @Ignore
```

Comment:
I think we should fix the UTs before adding JDK17, otherwise, it looks like JDK17 is supported but it might have weird quirks

Reply:
Definitely !

Reply:
Added the analysis of this failure here : https://github.com/apache/iceberg/pull/7391#discussion_r1173527238, both are the same UT just in different spark verions